tests/legacy-cli/e2e/utils/process.ts

```diff
@@ -96,39 +96,42 @@ function _exec(options: ExecOptions, cmd: string, args: string[]): Promise<Proce
   _processes.push(childProcess);
 
   // Create the error here so the stack shows who called this function.
+  const error = new Error();
+
+  // Return log info about the current process status
+  function envDump() {
+    return [
+      `ENV:${JSON.stringify(spawnOptions.env, null, 2)}`,
+      `STDOUT:\n${stdout}`,
+      `STDERR:\n${stderr}`,
+    ].join('\n\n');
+  }
 
-  return new Promise((resolve, reject) => {
+  return new Promise<ProcessOutput>((resolve, reject) => {
     let matched = false;
 
-    childProcess.on('exit', (error: any) => {
+    childProcess.on('exit', (code: number) => {
       _processes = _processes.filter((p) => p !== childProcess);
 
       if (options.waitForMatch && !matched) {
-        error = `Output didn't match '${options.waitForMatch}'.`;
+        reject(
+          `Process output didn't match - "${cmd} ${args.join(' ')}": '${
+            options.waitForMatch
+          }': ${code}...\n\n${envDump()}\n`,
```

Comment:
Note both the `code` (formerly `error`) and the `envDump()` were not here previously.